raft.go

```diff
@@ -505,6 +541,43 @@ func (r *Raft) leaderLoop() {
 		case <-r.leaderState.stepDown:
 			r.setState(Follower)
 
+		case future := <-r.leadershipTransferCh:
+			if r.getLeadershipTransferInProgress() {
+				r.logger.Printf("[DEBUG] raft: %s", ErrLeadershipTransferInProgress)
+				future.respond(ErrLeadershipTransferInProgress)
+				continue
+			}
+
+			// Track leadership status with the notify channel.
+			verify := &verifyFuture{}
+			verify.init()
+			r.leaderState.notify[verify] = struct{}{}
+			stopCh := make(chan error, 1)
+			doneCh := make(chan error, 1)
+
+			go func() {
+				select {
```

Comment:
also read from `shutdownCh` to make sure we don't hang around longer than we need to.

Reply:
This is accounted for because we have

```
defer func() { close(leftLeaderLoop) }()
```

which will close leftLeaderLoop and trigger the case.